snuba/datasets/generic_metrics_processor.py

```diff
@@ -86,24 +98,32 @@ def process_message(
         timestamp = _ensure_valid_date(datetime.utcfromtimestamp(message["timestamp"]))
         assert timestamp is not None, "Invalid timestamp"
 
+        try:
+            retention_days = enforce_retention(message["retention_days"], timestamp)
+        except EventTooOld:
+            return None
+
         keys = []
-        indexed_values = []
+        indexed_values: MutableSequence[int] = []
+        raw_values: MutableSequence[str] = []
         tags = message["tags"]
+        version = message.get("version", 1)
         assert isinstance(tags, Mapping), "Invalid tags type"
+        raw_values_index = self._get_raw_values_index(message)
 
         sorted_tag_items = sorted(tags.items())
         for key, value in sorted_tag_items:
-            assert key.isdigit() and isinstance(value, int), "Tag key/value invalid"
+            assert key.isdigit(), "Tag key invalid"
             keys.append(int(key))
-            indexed_values.append(value)
 
-        try:
-            retention_days = enforce_retention(message["retention_days"], timestamp)
-        except EventTooOld:
-            return None
-
-        raw_values_index = self._get_raw_values_index(message)
-        raw_values = [raw_values_index.get(str(v), "") for v in indexed_values]
+            if version == 1:
+                assert isinstance(value, int), "Tag value invalid"
```

Comment:
not really something to be fixed in this pr, but I wonder if it would be feasible to type out both messages like this:

```
class MessageV1(TypedDict):
    tags: Mapping[str, int]
    version: Literal[1]

class MessageV2(TypedDict):
    tags: Mapping[str, str]
    version: Literal[2]

Message = Union[MessageV1, MessageV2]
```